packages/compass-connections/src/stores/connections-store-redux.ts

```diff
@@ -1657,6 +1659,14 @@ const connectWithOptions = (
           return;
         }
 
+        // We're trying to optimise the initial Compass loading times here: to
+        // make sure that the driver connection pool doesn't immediately get
+        // overwhelmed with requests, we fetch instance info only once and then
+        // pass it down to telemetry and instance model. This is a relatively
+        // expensive dataService operation so we're trying to keep the usage
+        // very limited
+        const instanceInfo = await dataService.instance();
```

Comment:
This is somewhat a breaking change in Compass behavior as described in [COMPASS-7675](https://jira.mongodb.org/browse/COMPASS-7675) but I think it's probably okay to do so as we were already somewhat discussing doing this